### PR TITLE
feat(ui): DOMA-8921 updated Checkbox

### PR DESCRIPTION
**Pull request by Alllex202**
## Preview
https://github.com/open-condo-software/condo/assets/56914444/149278ca-a988-468f-8644-f6a2d2452a7e

### Before
Снимок экрана 2024-04-26 в 19 38 53

### After
Снимок экрана 2024-04-26 в 19 32 15

### CODE REVIEW

**Review by nomerdvadcatpyat — APPROVED** [review]

File: packages/ui/src/components/Checkbox/style.less [hunk 1]
```diff
@@ -6,110 +6,54 @@
...
 
+@condo-checkbox-focus-outline-width: @condo-global-border-width-default * 2;
+@condo-checkbox-focus-layer-margin: @condo-checkbox-focus-outline-width + @checkbox-border-width + 2px;
 @condo-checkbox-hidden-area: @condo-global-spacing-8;
 
 .condo-checkbox-wrapper {
   align-items: center;
 
+  * {
+    transition: all @condo-global-transition-duration-default;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Checkbox/style.less:23]: Curve? Like ease-in-out or smth...

Comment (SavelevMatthew) [reply packages/ui/src/components/Checkbox/style.less:23]: A bit dangerous selector to apply animations on all properties for all sub-element. Can we be more precise there?

File: packages/ui/src/components/Checkbox/style.less [hunk 2]
```diff
@@ -6,110 +6,54 @@
...
-
-    &-checked::after {
-      border: none;
-    }
-
-    &-checked > .condo-checkbox-inner {
-      box-sizing: content-box;
-      background: @condo-global-color-gray-5 border-box;
-      border-color: transparent;
+      border-color: @condo-global-color-gray-5 !important;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Checkbox/style.less:36]: transparent was working without important flag, why gray doesn't?)

File: packages/ui/src/components/Checkbox/style.less [hunk 3]
```diff
@@ -125,38 +69,49 @@
     }
   }
 
-  &:hover > .condo-checkbox > .condo-checkbox-inner {
-    border-color: @condo-global-color-gray-7;
+  &-disabled .condo-checkbox-inner {
+    opacity: 0.5;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Checkbox/style.less:73]: use token here

File: packages/ui/src/components/Checkbox/style.less [hunk 4]
```diff
@@ -125,38 +69,49 @@
...
-  &:hover > .condo-checkbox-checked > .condo-checkbox-inner::before {
-    border-radius: @checkbox-inner-border-radius;
-  }
+  .condo-checkbox.condo-checkbox-indeterminate > .condo-checkbox-inner {
+    background: @condo-global-color-brand-gradient-1;
 
-  .condo-checkbox-indeterminate > .condo-checkbox-input:focus + .condo-checkbox-inner::before {
-    .condo-checkbox-inner-layer(@condo-global-color-brand-gradient-1, @checkbox-inner-border-radius);
+    &::after {
+      width: 14px;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Checkbox/style.less:80]: why this size? Is it related to font? or why `x % 4 !== 0`

File: packages/ui/src/components/Checkbox/style.less [hunk 5]
```diff
@@ -125,38 +69,49 @@
...
+    &::after {
+      top: @condo-checkbox-checkmark-top - 1;
+      left: @condo-checkbox-checkmark-left;
+      box-sizing: content-box;
+      width: @condo-checkbox-checkmark-width;
+      height: @condo-checkbox-checkmark-height;
+      border-width: 3px;
+      border-image-source: linear-gradient(45deg, #4cd174 0%, #6db8f2 100%);
+      border-image-slice: 1;
+      transform: rotate(48deg) scale(1) translate(-50%, -50%);
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Checkbox/style.less:99]: why 48? and -1 above

File: packages/ui/src/components/Checkbox/style.less (hunk 2 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:36]: Previously, there was also important flag, but only in a different place

File: packages/ui/src/components/Checkbox/style.less (hunk 5 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:99]: This is an already existing style, I did not change it)

File: packages/ui/src/components/Checkbox/style.less (hunk 4 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:80]: Will be 12px, does not related on typography

**Review by toplenboren — APPROVED** [review]

File: packages/ui/src/components/Checkbox/style.less (hunk 1 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:23]: was added mixin for transition

File: packages/ui/src/components/Checkbox/style.less (hunk 2 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:36]: fixed

File: packages/ui/src/components/Checkbox/style.less (hunk 5 above)
Comment (Alllex202) [reply packages/ui/src/components/Checkbox/style.less:99]: There is no need to change the degrees, they affect the tilt of checkmark

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**